tstore/tsdf/tsdf.py

```diff
@@ -1,28 +1,137 @@
 """Module defining the TSDF abstract wrapper for a dataframe of TSArray objects."""
 
-from tstore.backend import DataFrame, PandasDataFrame
+from typing import TYPE_CHECKING
+
+from tstore.backend import (
+    Backend,
+    DaskDataFrame,
+    DataFrame,
+    PandasDataFrame,
+    PolarsDataFrame,
+    PyArrowDataFrame,
+    remove_dataframe_index,
+)
+from tstore.tsdf.ts_dtype import TSDtype
 from tstore.tswrapper.tswrapper import TSWrapper
 
+if TYPE_CHECKING:
+    # To avoid circular imports
+    from tstore.tslong.tslong import TSLong
+    from tstore.tswide.tswide import TSWide
+
 
 class TSDF(TSWrapper):
-    """Abstract wrapper for a dataframe of TSArray objects."""
+    """Abstract wrapper for a DataFrame of TSArray objects."""
+
+    def __init__(
+        self,
+        df: DataFrame,
+        id_var: str,
+    ) -> None:
+        """Wrap a DataFrame of TSArrays as a TSDF object.
+
+        Args:
+            df (DataFrame): DataFrame to wrap.
+            id_var (str): Name of the column containing the identifier variable.
+        """
+        if not isinstance(df, PandasDataFrame):
+            raise TypeError(
+                "The input dataframe must be a Pandas DataFrame. Inner TS objects can contain dataframes of any type.",
+            )
+
+        df = remove_dataframe_index(df)
+        super().__init__(df)
+        # Set attributes using __dict__ to not trigger __setattr__
+        self.__dict__.update(
+            {
+                "_tstore_id_var": id_var,
+            },
+        )
 
     def __new__(cls, *args, **kwargs) -> "TSDF":
         """When calling TSDF() directly, return the appropriate subclass."""
         if cls is TSDF:
-            df = kwargs.get("df", args[0])
-            return TSDF.wrap(df)
+            return TSDF.wrap(*args, **kwargs)
 
         return super().__new__(cls)
 
+    def change_backend(self, new_backend: Backend) -> "TSDF":
+        """Return a new TSDF object with dataframes wrapped in internal TS objects converted to a different backend."""
+        df = self._obj.copy()
+        ts_cols = get_ts_columns(df)
+
+        df[ts_cols] = df[ts_cols].applymap(lambda ts_obj: ts_obj.change_backend(new_backend))
+        df[ts_cols] = df[ts_cols].astype(TSDtype)
```

Comment:
Interesting. This is a single-thread loop. 
Maybe we should implement a parallel loop for such operations. 
`astype(TSDtype)` is necessary? Does not recognize it already?